hooks/rnc-format-validate.sh

```diff
@@ -0,0 +1,19 @@
+#!/usr/bin/env bash
+
+set -e
+
+# assert the trang command exists
+if ! [ -x "$(command -v trang)" ]; then
+    echo 'trang command not found; please install'
+    exit 1
+fi
+
+# iterate through changed rnc files
+for rncfile in $(git diff --cached --name-only | egrep '\.rnc$')
+do
+  rngfile="${rncfile//+(*\/|.*)}.rng"
+  trang $rncfile $rngfile && trang $rngfile $rncfile
+  rm -f $rngfile
+  git add $rncfile
```

Comment:
Typically with `pre-commit`, the idea is the check runs, and if it fails, the user manually fixes the files, then does `git add`.

For the typical case, this make sense. But I don't think it does for this case. So the script does for you.

In essence, it auto fixes your formatting problems.